packages/suite-desktop-core/src/modules/bioAuthModule.ts

```diff
@@ -24,104 +18,144 @@ const loadWin = async ({ mainWindowProxy }: Dependencies) => {
+                        success: false as const,
+                        message: serializeError(error),
+                    };
+                }
+            },
+            {
+                timeout: 60_000,
+            },
+        ).catch(() => ({
+            success: false as const,
+            message: 'timeout',
+        }));
     });
 
-    ipcMain.on('bio-auth/request-availability', async () => {
-        try {
-            const available = await winHello.isHelloAvailable();
-
-            if (!available) {
-                logger.info('bioAuth', 'WIN: passport is not available');
-            }
-            mainWindowProxy.getInstance()?.webContents.send('bio-auth/is-available', available);
-        } catch (error) {
-            logger.info('bioAuth', `WIN: bioAuth isAvailable failed: ${error}`);
-
-            mainWindowProxy.getInstance()?.webContents.send('bio-auth/is-available', false);
-        }
+    ipcMain.handle('bio-auth/is-bio-auth-available', ipcEvent => {
+        validateIpcMessage(ipcEvent);
+
+        return scheduleAction(
+            async () => {
+                try {
+                    const available = await winHello.isHelloAvailable();
+
+                    if (!available) {
+                        logger.info('bioAuth', 'WIN: passport is not available');
+                    }
+
+                    return available;
+                } catch (error) {
+                    logger.info('bioAuth', `WIN: bioAuth isAvailable failed: ${error}`);
+
+                    return false;
+                }
+            },
+            {
+                timeout: 60_000,
+            },
+        ).catch(() => false);
     });
 
     return {
         destroy: () => winHello.destroy(),
     };
 };
 
-const loadMac = ({ mainWindowProxy }: Dependencies) => {
+const loadMac = () => {
     const { logger } = global;
-    ipcMain.on('bio-auth/request', async (_, params) => {
-        try {
-            await systemPreferences.canPromptTouchID();
-        } catch (error) {
-            logger.info('bioAuth', `MAC: bioAuth canPromptTouchID failed: ${error}`);
-            mainWindowProxy.getInstance()?.webContents.send('bio-auth/validated', {
-                success: false,
-                message: serializeError(error),
-            });
-
-            return;
-        }
-
-        try {
-            await systemPreferences.promptTouchID(params.message ?? PROMPT_REASON);
-            mainWindowProxy.getInstance()?.webContents.send('bio-auth/validated', {
-                success: true,
-            });
-
-            return;
-        } catch (error) {
-            logger.info('bioAuth', `MAC: bioAuth validation failed: ${error}`);
-            mainWindowProxy.getInstance()?.webContents.send('bio-auth/validated', {
-                success: false,
-                message: serializeError(error),
-            });
-        }
+    ipcMain.handle('bio-auth/validate-bio-auth', (ipcEvent, params) => {
```

Comment:
FYI: `ipcMain.handle` was here earlier, but I reverted it, coz I perceived it unreliable `24fa65c9335f69c302ff3f57233d68cab1a64c63`

Although, I can't find specific reason for the revert.